src/colorize.cr

```diff
@@ -401,12 +412,12 @@ struct Colorize::Object(T)
         printed = true
       end
 
-      unless mode_is_default
+      unless no_mode
         # Can't reuse MODES constant because it has bold/bright duplicated
-        {% for name in %w(bold dim underline blink reverse hidden) %}
-          if mode.bits_set? MODE_{{name.upcase.id}}_FLAG
+        {% for mode in %w(Bold Dim Underline Blink Reverse Hidden) %}
```

Comment:
```suggestion
        Mode.values.each do |mode|
```

Reply:
I assume you mean
`{% for mode in Mode.constants.reject { |constant| constant == "Bright" } %}`? However that's even longer than `{% for mode in %w(Bold Dim Underline Blink Reverse Hidden) %}` so maybe this little boilerplate code is fine.

Reply:
No, I meant a runtime loop. I don't think we need a macro for this. The only reason for the macro is to use `MODE_*` constants. But this could just be a simple mapping from `Mode` to mode id.

Reply:
So a hash? I'm not sure if this little change is really worth the slow down. I would like to keep this as performant as possible and avoid hash look-ups especially in this method which is pretty frequently called.
Unfortunately I can't use a named tuple either because only symbols and strings are accepted as keys, not constants like `Mode::Bold`.

Reply:
Nobody's suggesting a hash. Simpy use `case ... when`